openshift/helper/validate.py

```diff
@@ -0,0 +1,35 @@
+# validate helper allows openshift client to validate kubernetes resource definitions
+# Note that it is expected that the caller catches the lack of kubernetes_validate
+
+import kubernetes_validate
+
+
+def validate(resource, client=None, version=None, strict=False):
+    """validate checks a kubernetes resource definition
+
+    Args:
+        resource (dict): resource definition
+        client (object): openshift dynamic client used to obtain version
+        version (str): version of kubernetes to validate against
+        strict (bool): whether unexpected additional properties should be considered errors
+
+    Returns:
+        warnings (list), errors (list): warnings are missing validations, errors are validation failures
+    """
+
+    errors = list()
+    warnings = list()
+    try:
+        if version is None:
+            if client:
+                version = client.version['kubernetes']['gitVersion']
+            else:
+                version = kubernetes_validate.latest_version()
+        kubernetes_validate.validate(resource, version, strict)
+    except kubernetes_validate.utils.ValidationError as e:
+        errors.append("resource validation error at %s: %s" % ('.'.join([str(item) for item in e.path]), e.message))
```

Comment:
I think `e.message` is making linting for python3 fail, I guess it was removed. The message is

```
B306 `BaseException.message` has been deprecated as of Python 2.6 and is removed in Python 3. 
Use `str(e)` to access the user-readable message. 
Use `e.args` to access arguments passed to the exception.
```

Reply:
@fabianvf - `e.message` was removed from BaseException, sure. But ValidationError sets `message` explicitly.

https://python-jsonschema.readthedocs.io/en/latest/errors/#jsonschema.exceptions.ValidationError

If I test almost identical code in python3 from kubernetes-validate codebase, it works fine:
```
PYTHONPATH=src python3 src/kubernetes_validate/__main__.py --strict examples/kuard-extra-property.yaml 
ERROR examples/kuard-extra-property.yaml did not validate against version 1.11.0: spec.selector: Additional properties are not allowed ('unwanted' was unexpected)
```